playbooks/06-deploy-va.yml

```diff
@@ -0,0 +1,124 @@
+      vars:
+        _ifaces_vars: >-
+          {{
+            hostvars['controller-0'].ansible_interfaces |
+            map('regex_replace', '^(.*)$', 'ansible_\1')
+          }}
+        _controller_host: "{{ hostvars['controller-0'].ansible_host }}"
+        _ipv4_network_data: >-
+          {{
+            hostvars['controller-0'] | dict2items |
+            selectattr('key', 'in', _ifaces_vars) |
+            selectattr('value.ipv4.address', 'defined') |
+            selectattr('value.ipv4.address', 'equalto', _controller_host) |
+            map(attribute='value.ipv4') | first | default({})
+          }}
+      ansible.builtin.set_fact:
+        cifmw_ci_gen_kustomize_values_ssh_authorizedkeys: >-
+          {{ lookup('file', '~/.ssh/authorized_keys', rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_ssh_private_key: >-
+          {{ lookup('file', '~/.ssh/id_cifw', rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_ssh_public_key: >-
+          {{ lookup('file', '~/.ssh/id_cifw.pub', rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_migration_pub_key: >-
+          {{ lookup('file', _nova_key.filename ~ '.pub', rstrip=False)}}
+        cifmw_ci_gen_kustomize_values_migration_priv_key: >-
+          {{ lookup('file', _nova_key.filename, rstrip=False) }}
+        cifmw_ci_gen_kustomize_values_sshd_ranges: >-
+          {{
+            [cifmw_networking_env_definition.networks.ctlplane.network_v4] +
+            (
+              [
+                 _ipv4_network_data.network + '/' + _ipv4_network_data.prefix
+              ]
+            ) if (_ipv4_network_data | length > 0) else []
+          }}
+
+    - name: Load architecture automation file
+      register: _automation
+      ansible.builtin.slurp:
+        path: "{{ cifmw_architecture_automation_file }}"
+
+    - name: Prepare automation data
+      vars:
+        _parsed: "{{ _automation.content | b64decode | from_yaml }}"
+      ansible.builtin.set_fact:
+        cifmw_deploy_va_steps: >-
+          {{ _parsed['vas'][cifmw_architecture_va_scenario] }}
+
+    - name: Check requirements
+      ansible.builtin.import_role:
+        name: kustomize_deploy
+        tasks_from: check_requirements.yml
+
+    - name: Deploy OSP operators
+      ansible.builtin.import_role:
+        name: kustomize_deploy
+        tasks_from: install_operators.yml
+
+    - name: Configure Storage Class
+      ansible.builtin.include_role:
+        name: ci_local_storage
+
+    - name: Execute deployment steps
+      ansible.builtin.include_role:
+        name: kustomize_deploy
+        tasks_from: execute_step.yml
+      loop: "{{ cifmw_deploy_va_steps.stages }}"
+      loop_control:
+        label: "{{ stage.path }}"
+        loop_var: stage
+        index_var: stage_id
+
+    - name: Extract and install OpenStackControlplane CA
+      ansible.builtin.include_role:
+        role: install_openstack_ca
+
+    - name: Run nova host discover process
```

Comment:
I'm sure all the VAs/DTs will deploy a dataplane, but, if at somepoint we want to use this for CI and deploy a cropped OSP with only control-plane, we won't run this.

Reply:
maybe this should be part of a post_run hook in the latest stage as described in the architecture automation? @fultonj @abays any thoughts?

Reply:
So add a `post_stage_run` to [1] like this?
```
        post_stage_run:
          - name: Run nova host discover process
            type: command?
            ?: oc rsh -n openstack nova-cell0-conductor-0 nova-manage cell_v2 discover_hosts --verbose
```

[1] https://github.com/openstack-k8s-operators/architecture/blob/main/automation/vars/default.yaml#L43

Reply:
or maybe a validation? Not really happy to inject a command support in hooks, to be honest. The "playbook" approach sounds overkill - sooo... bleh. TBD.

Reply:
TODO: find a better way to trigger that command - for now we keep it as is, and will revisit.